src/erlcloud_mon.erl

```diff
@@ -216,33 +217,111 @@ put_metric_data(Namespace, MetricName, Value, Unit, Timestamp) ->
          ),
     mon_simple_query(Config, "PutMetricData", Params).
 
+
+%%------------------------------------------------------------------------------
+%% @doc CloudWatch API - GetMetricStatistics - Easy average version
+%% Gets average stats at 60 second intervals for the given metric on the given instance for the given interval
+%% @end
+%%------------------------------------------------------------------------------
+-spec get_metric_statistics(
+        MetricName  ::string(),
+        StartTime   ::datetime(),
+        EndTime     ::datetime(),
+        InstanceId  ::string()
+       ) -> term().
+
+get_metric_statistics(
+  MetricName,
+  StartTime,
+  EndTime,
+  InstanceId) ->
+          get_metric_statistics(
+              "AWS/EC2",
+              MetricName,
+              StartTime,
+              EndTime,
+              60,
+              "",
+              ["Average","Maximum"],
+              [{'InstanceId', InstanceId}]).
+
 %%------------------------------------------------------------------------------
 %% @doc CloudWatch API - GetMetricStatistics
 %% [http://docs.amazonwebservices.com/AmazonCloudWatch/latest/APIReference/index.html?API_GetMetricStatistics.html]
-%% @end
+%%
+%% USAGE:
+%%
+%%  erlcloud_mon:get_metric_statistics(
+%%    "AWS/EC2",
+%%    "CPUUtilization",
+%%    {{2016, 06, 29}, {0, 0, 0}},
+%%    {{2016, 06, 29}, {0, 30, 0}},
+%%    60,
+%%    "Percent",
+%%    ["Average", "Maximum"],
+%%    [{"InstanceType", "t2.micro"}]).
+%%                                     
+%% @end 
 %%------------------------------------------------------------------------------
 -spec get_metric_statistics(
         Namespace   ::string(),
         MetricName  ::string(),
-        StartTime   ::string(),
-        EndTime     ::string(),
+        StartTime   ::datetime(),
+        EndTime     ::datetime(),
         Period      ::pos_integer(),
         Unit        ::string(),
         Statistics  ::[string()],
-        Dimensions  ::[string()]
+        Dimensions  ::[{string(), string()}]
                       ) -> term().
 
 get_metric_statistics(
-  _Namespace,
-  _MetricName,
-  _StartTime,
-  _EndTime,
-  _Period,
-  _Unit,
-  _Statistics,
-  _Dimensions
+  Namespace,
+  MetricName,
+  StartTime,
+  EndTime,
+  Period,
+  Unit,
+  Statistics,
+  Dimensions
  ) ->
-    todo.
+    Config = default_config(),
+    Params =
+          [
+           {"Namespace",  Namespace},
+           {"MetricName", MetricName},
+           {"StartTime",  format_timestamp(StartTime)},
+           {"EndTime",    format_timestamp(EndTime)},
+           {"Period",     Period},
+           [{"Unit",      Unit} || Unit=/=undefined, Unit=/=""]
```

Comment:
Ah I get it, thanks for the explanation. It didn't matter if the param values were lists anyway, since flatten doesn't go inside tuples it looks like.

It worked the way I had it for some reason. It seems like it shouldn't. Instead of tracing down why I just added the lists:flatten.